src/ngx_http_lua_util.c

```diff
@@ -3980,4 +3985,84 @@ ngx_http_lua_get_raw_phase_context(lua_State *L)
     return 1;
 }
 
+
+ngx_http_cleanup_t *
+ngx_http_lua_cleanup_add(ngx_http_request_t *r, size_t size)
+{
+    ngx_http_cleanup_t  *cln;
+    ngx_http_lua_ctx_t  *ctx;
+
+    r = r->main;
```

Comment:
Oh, we should add a test case to exercise this special subrequest case stated above as well :)